DolbyIO/Source/Private/DolbyIOVideoSink.cpp

```diff
@@ -54,30 +53,47 @@ namespace DolbyIO
 		else
 		{
 			DLB_UE_LOG("Creating texture: %dx%d", Width, Height);
-			Fence.BeginFence();
 		}
-
-		Region.Width = Width;
-		Region.Height = Height;
-		Buffer.SetNumUninitialized(Width * Height * Stride);
 		Texture = UTexture2D::CreateTransient(Width, Height, PF_B8G8R8A8);
 		Texture->AddToRoot();
-		Texture->UpdateResource();
 
 		for (UMaterialInstanceDynamic* Material : Materials)
 		{
 			UpdateMaterial(Material);
 		}
 	}
 
+#if ENGINE_MAJOR_VERSION == 4
+#define PLATFORM_DATA PlatformData
+#else
+#define PLATFORM_DATA GetPlatformData()
+#endif
+
 	void FVideoSink::Convert(dolbyio::comms::video_frame& VideoFrame)
 	{
+		constexpr int Stride = 4;
+		struct FTextureBuffer
+		{
+			FTextureBuffer(UTexture2D* Texture) : Texture(Texture) {}
+			~FTextureBuffer()
+			{
+				Texture->PLATFORM_DATA->Mips[0].BulkData.Unlock();
+				Texture->UpdateResource();
+			}
+			operator uint8_t*()
+			{
+				return reinterpret_cast<uint8_t*>(Texture->PLATFORM_DATA->Mips[0].BulkData.Lock(LOCK_READ_WRITE));
```

Comment:
This works right but doesn't look right. I'd rather have this lock in the constructor than in a cast operator. You would need an additiona uint8_t* member but it would be clearer.

Reply:
we use it once so for me this fine (I considered putting sth like `SingleUse` in its name to avoid misuse), but if you insist,  I can add one more member to make it usable many times.